compiler/function-colors.cpp

```diff
@@ -0,0 +1,116 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2020 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "function-colors.h"
+
+using namespace function_palette;
+
+const std::map<std::string, color_t> str2color_type = {
+  {"*",                 color_t::any},
+  {"highload",          color_t::highload},
+  {"no-highload",       color_t::no_highload},
+  {"ssr",               color_t::ssr},
+  {"ssr-allow-db",      color_t::ssr_allow_db},
+  {"has-db-access",     color_t::has_db_access},
+  {"api",               color_t::api},
+  {"api-callback",      color_t::api_callback},
+  {"api-allow-curl",    color_t::api_allow_curl},
+  {"has-curl",          color_t::has_curl},
+  {"message-internals", color_t::message_internals},
+  {"message-module",    color_t::message_module},
+  {"danger-zone",       color_t::danger_zone},
+};
+
+color_t function_palette::parse_color(const std::string &str) {
+  const auto it = str2color_type.find(str);
+  const auto found = it != str2color_type.end();
+  return found ? it->second : color_t::none;
+}
+
+std::string function_palette::color_to_string(color_t color) {
+  for (const auto &pair : str2color_type) {
+    if (pair.second == color) {
+      return pair.first;
+    }
+  }
+  return "";
+}
+
+const ColorContainer::ColorsRaw &ColorContainer::colors() {
+  return this->data;
```

Comment:
We usually don't use explicit `this->` member access in a new code.

Reply:
Thanks, corrected.